Source/Core/Duality/Utility/Coroutines/CoroutineSceneExtensions.cs

```diff
@@ -0,0 +1,36 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using Duality.Resources;
+using Duality.Utility.Coroutines;
+
+namespace Duality.Utility.Coroutines
+{
+	public static class CoroutineSceneExtensions
+	{
+		/// <summary>
+		/// Starts a coroutine and registers it in the scene's CoroutineManager
+		/// </summary>
+		/// <param name="scene">The Scene</param>
+		/// <param name="method">The coroutine implementation</param>
+		/// <param name="name">The name of the coroutine, optional</param>
+		/// <returns>The created corutine</returns>
+		public static Coroutine StartCoroutine(this Scene scene, IEnumerable<WaitUntil> method, string name = null)
+		{
+			return scene.StartCoroutine(method.GetEnumerator(), name);
```

Comment:
Make the CoroutineManager itself be able accept a IEnumerable. Will help testing and clean up these utility methods.

Reply:
Won't you end up with one extra method to maintain in the end?
`Scene.Start(IEnumerable) => CoroutineManager.Start(IEnumberable)`
+
`Scene.Start(IEnumerator) => CoroutineManager.Start(IEnumerator)`

Reply:
yes but the utility methods will contain basically no logic

Reply:
do we need to provide a API that accepts a IEnumerator btw? Isnt IEnumerable alone enough?

Reply:
Eeeeeh.. I like to keep my options open as you might have noticed 😄 don't have a strong opinion towards removing the IEnumerator version though.

Reply:
I cant think of a use case where you would pass in the IEnumerator directly so I think we should remove it

Reply:
Off it goes